open-craft/xblock-chat
# Fix layout in the Android app

This fixes a couple of issues found on the app running on Android Emulator.

1. The user's avatar box was being trimmed on the right. – I fixed it by adjusting the padding for mobile only.
2. The multiple choice answer buttons were being trimmed at the bottom, because the main chat area wasn't fully expanding in height. – I fixed this by moving the subject out of the `.chat-block` div, gave it a maximum height to be able to account for it when calculating the height of the chat area, as the subject div wasn't being considered at all and it was pushing down the `.main-area` with every new line of text.

**Testing**
1. Add the xblock chat to a unit
2. Edit the xblock's dialogue script to also display 3+ possible answers for some questions (as the bug was behaving differently based on the number of choices)
3. Go through the entire dialogue with the bot and make sure nothing is trimmed at the bottom
4. Test again with different subject lengths and also with no subject at all

Before/after my changes:
<img width="336" alt="screen shot 2017-08-30 at 23 11 37" src="https://user-images.githubusercontent.com/1135997/29965053-701d5bce-8f0c-11e7-9640-4cd4fafc1c21.png"><img width="331" alt="screen shot 2017-09-01 at 12 02 56" src="https://user-images.githubusercontent.com/1135997/29965350-85562d30-8f0d-11e7-9947-d78e02ddf3ca.png">

**Author notes and concerns**
Initially I was looking into flexbox for a possible solution, but then I noticed it was actually removed in favour of the current approach. I am not sure if this is the preferred fix, I tried to be as consistent as possible with the existing code. All the changes are applied to the `.course-wrapper.chromeless` class so they don't affect desktop browsers.

I tested this in all browsers (except for IE10) and on the Android Emulator. Unfortunately I haven't been able to build the iOS app to test there as well (and getting the Android one to work was time-consuming enough), so hopefully the reviewer can help me with that.
